org.jcryptool.core.introduction/src/org/jcryptool/core/introduction/views/AlgorithmInstruction.java

```diff
@@ -400,6 +398,29 @@ public void widgetDefaultSelected(SelectionEvent e) {
 
 	}
 
+	private int[] computeSlideshowSizeHint() {
+		Rectangle parentSize = content.getBounds();
+		float aspectRatio = getCurrentSlideAspectRatio();
+		float parentAspectRatio = (float) parentSize.width / (float) parentSize.height;
+		int adaptedWidth = parentSize.width;
+		int adaptedHeight = parentSize.height;
+		if(adaptedWidth <= 0 || adaptedHeight <= 0) {
+			adaptedHeight = 10; // TODO: handle better?
+			adaptedWidth = 10;
+		}
+		if (aspectRatio > parentAspectRatio) { // broader than allowed -> adapt height to match parent width
+			adaptedHeight = (int) Math.round(adaptedWidth / aspectRatio);
+		} else {                               // other way around
+			adaptedWidth = (int) Math.round(adaptedHeight * aspectRatio);
+		}
+		int[] adaptedSize = new int[] {adaptedWidth, adaptedHeight};
+		return adaptedSize;
+	}
+
+	protected float getCurrentSlideAspectRatio() {
+		return 16.0f / 9.0f; //TODO: dynamically get aspect ratio from displayed img
+	}
+
```

Comment:
This is just me being a bit ~lazy~ efficient, please retrieve the actual dimensions of the displayed picture. It's best to avoid reloading an image from file or something like that for this little piece of information (cache it somewhere at first image data retrieval would be my approach).

Reply:
I think I will retrieve the ratio of the current image by taking a look at its ImageData. This shouldn't afffect the performance to much, because I have the images stored already in an array (I don not need to load the image from a file). I would avoid using one ration for all images, because the size of the images can differ.